runtime/parser/declaration.go

```diff
@@ -1410,6 +1397,21 @@ func parseSpecialFunctionDeclaration(
 		declarationKind = common.DeclarationKindPrepare
 	}
 
+	if returnTypeAnnotation != nil {
```

Comment:
Do we want to explicitly disallow return type annotations for special functions? I don't think it hurts readability to add a `: Void`, especially if `Void` return annotations are used elsewhere in a codebase.

Reply:
We already currently reject explicit type annotations for special functions, but do so implicitly by not parsing any / looking for one: The old parsing code only parses the parameter list and the (potentially) optional body.

we could allow an explicit `Void` return type annotation in the future, but I do not see any value in it. A user learning the language might assume that it could return something other than `Void`, which is not supported or makes sense for any of the special functions (`init`, `destroy`, etc.)

Reply:
Hmm, seems that we should move this validation to the checker instead then. Even if a user assumes special functions can have different return types, they'll immediately get feedback in the form of a type error (which we can specialize to that scenario too, e.g. `Wrong type for init() function, expected Void`). It's not like special functions have any other signifiers other than the docs anyway